hs_core/page_processors.py

```diff
@@ -34,6 +35,16 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
 
     metadata_status = _get_metadata_status(content_model)
 
+    relevant_tools = []
+    for tool in ToolResource.objects.all():
+        if len(tool.metadata.res_types.all())>0:
```

Comment:
`len(...all())` does a query for the entire contains of the model and then, only once its all been pulled from the database, is counted on the Python side. Use `tool.metadata.res_type.all().count()` to get this directly from the database. It will be much more efficient.